crates/core/tedge/src/cli/diag/cli.rs

```diff
@@ -0,0 +1,65 @@
+use crate::cli::diag::collect::DiagCollectCommand;
+use crate::command::BuildCommand;
+use crate::command::Command;
+use crate::ConfigError;
+use camino::Utf8PathBuf;
+use tedge_config::models::SecondsOrHumanTime;
+use tedge_config::TEdgeConfig;
+use tedge_config::TEdgeConfigLocation;
+use time::format_description;
+use time::OffsetDateTime;
+
+#[derive(clap::Subcommand, Debug)]
+pub enum TEdgeDiagCli {
+    /// Collect diagnostic logs
+    Collect {
+        /// Directory where diagnostic plugins are stored
+        #[clap(long, default_value = "/etc/tedge/diag-plugins")]
+        plugin_dir: Utf8PathBuf,
```

Comment:
`tedge diag collect` currently does nothing if this directory is empty, while I would expect at least the regular tedge logs to be collected.

Reply:
Hmm the regular tedge logs collection (I guess you mean under `/var/log/tedge`) is planned to be done by a plugin (in my example, `04_workflow.sh`). Since the script is just copying files, we can implement in Rust. But not sure which way is better.